server/main.py

```diff
@@ -98,26 +103,24 @@ def disconnect(self, client: Client, room: str) -> None:
             client: The Client to which the connection belongs.
             room: The room from which the client will be disconnected.
         """
-        self._rooms[room]["clients"].remove(client)
+        self._rooms[room_name]["clients"].remove(client)
 
-        if len(self._rooms[room]["clients"]) == 0:
-            del self._rooms[room]
+        if len(self._rooms[room_name]["clients"]) == 0:
+            del self._rooms[room_name]
 
-    async def broadcast(self, data: dict, room: str, everyone: bool = False) -> None:
+    async def broadcast(self, data: dict, room_name: str, sender: Client | None = None) -> None:
```

Comment:
Is there a situation where we wouldn't pass in the client?

Reply:
Only asking because we should probably bring in a static type checker at some point which would give an error here about how sender can be None or something like that 😛

Reply:
> Is there a situation where we wouldn't pass in the client?

When you want to send the event to everyone in the room, including the sender (basically setting `everyone` to `True`).

Reply:
> Only asking because we should probably bring in a static type checker at some point which would give an error here about how sender can be None or something like that stuck_out_tongue

Fixed in https://github.com/Vthechamp22/kindly-kappa/pull/24/commits/69fc69a5a8b2080477eceec888fd43870f6a8a0a